lib/rouge/lexers/ruby.rb

```diff
@@ -183,7 +183,7 @@ def self.detect?(text)
 
         mixin :strings
 
-        rule %r/(?:#{keywords.join('|')})\b/, Keyword, :expr_start
+        rule %r/(?:#{keywords.join('|')})(?=[ ]*\b|\()/, Keyword, :expr_start
```

Comment:
Do we want to match the word boundary if there are spaces? Won't that mean that strings that have non-word characters (eg. `[`) aren't matched if they come after spaces but are matched if they don't come after spaces?

Reply:
To be honest, this regex is a hack to work with Ruby lexer's `keywords` array.

The word boundary matches at the position between `\w` and `\W`. Ideally, we wouldn't need to match or test for whitespace if we're also testing for word boundary.
But in this case, our *keyword* is `/defined\?/`. Word boundary is either before the first `d` and after the last `d`. Therefore, `/defined\?\b/` would never match.

I'll define a new `rule` to eliminate this cognitive overhead.

Reply:
Ahhhh, yeah, I get it now. I like the consistency of having `defined?` be included in `keywords` but it does simplify the (human) parsing of the rules to split it out. 

At the risk of making this an ideal example of bikeshedding, do you think it might be better to use the following rule:

```ruby
rule %r/(?:#{keywords.join('|')})(?=\W|$)/, Keyword, :expr_start
```

Reply:
*Au contraire*, your suggestion is the more apt change. :smiley:

Reply:
It was your explanation :)